core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -110,7 +172,9 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging {
 
     def abortQueuedWorkflows(submissionId: UUID) = {
       datasource.inTransaction { dataAccess =>
-        dataAccess.workflowQuery.batchUpdateWorkflowsOfStatus(submissionId, WorkflowStatuses.Queued, WorkflowStatuses.Aborted)
+        workflowStatusCounter(WorkflowStatuses.Aborted).countDBResult {
```

Comment:
Yeah, I see your point. I guess I was just hesitant to add side effects to what is now pure database-affecting code. But it might make the code easier to work with. Let me take a stab at moving the counters to `WorkflowComponent` (there are still multiple methods that need updating) and see how it looks.